app/workers/generic_worker.rb

```diff
@@ -0,0 +1,14 @@
+class GenericWorker
+
+  include Sidekiq::Worker
+
+  def perform(klass_name, klass_method, *method_args)
+    klass = klass_name.constantize
```

Comment:
We should also pass User.current to the worker so we can set User.current  before calling the background method
User.current  = passed_variable
......set the logic...
User.current  = nil

Reply:
@melsawy, a question: Why do we need `User.current` here?

Reply:
We log the revisions for tag creation in case User.current exists otherwise no logs for this action

Reply:
@melsawy another question for when you are back:
The `GenericWorker` is as the name says, so, do we always want to set `User.current`? Does it make sense to always set `User.current`?

I wasn't sure about that, so I wrote it like this for now, but I'm not too sure if it makes sense
https://github.com/meedan/check-api/blob/353612139eca53830ed062ac8612cc03b36a9a11/app/workers/generic_worker.rb#L5-L18

Reply:
Yes make sense to always set `User.current` and I suggest the following code for User.current

   if options
      user_id = options.delete(:user_id) if options.key?(:user_id)
      current_user = User.current
      User.current = User.find_by_id(user_id)
      klass.public_send(klass_method, **options)
      User.current = current_user
    else
      klass.public_send(klass_method)
    end